allennlp/tests/training/metrics/pearson_correlation_test.py

```diff
@@ -6,55 +6,82 @@
 from allennlp.common.testing import AllenNlpTestCase
 from allennlp.training.metrics import PearsonCorrelation
 
+def pearson_corrcoef(predictions, labels, fweights=None):
+    covariance_matrices = np.cov(predictions, labels, fweights=fweights)
+    denominator = np.sqrt(covariance_matrices[0, 0] * covariance_matrices[1, 1])
+    if np.around(denominator, decimals=5) == 0:
+        expected_pearson_correlation = 0
+    else:
+        expected_pearson_correlation = covariance_matrices[0, 1] / denominator
+    return expected_pearson_correlation
+
 
 class PearsonCorrelationTest(AllenNlpTestCase):
     def test_pearson_correlation_unmasked_computation(self):
         pearson_correlation = PearsonCorrelation()
         batch_size = 100
         num_labels = 10
-        predictions = np.random.randn(batch_size, num_labels).astype("float32")
-        labels = 0.5 * predictions + np.random.randn(batch_size, num_labels).astype("float32")
+        predictions_1 = np.random.randn(batch_size, num_labels).astype("float32")
+        labels_1 = 0.5 * predictions_1 + np.random.randn(batch_size, num_labels).astype("float32")
+
+        predictions_2 = np.random.randn(1).repeat(num_labels).astype("float32")
```

Comment:
Can you give a comment above this for why you are doing two separate tests?

Reply:
I am very happy to see your review. The reason for fixing this bug is that in some input data, a batch's `predictions` or `labels` may be exactly the same. Then when the pearson_correlation.py is called, it will appear that the `predictions_variance` or `labels_variance` of the batch is 0 when calculating the `variance`, resulting in:

`ZeroDivisionError: float division by zero`

The two tests are separated because: in most cases, the data is constructed like predictions_1; but in a few cases, for example: predictions_2, the data of such a batch is exactly the same.

Reply:
Yes, I get the error, I was just wanting some more description _in the code_, so that if someone comes back to this several months down the line, it'll be more obvious why there are two sets of predictions and labels.